[Docs] Remove Self hosting runbook + Helm references

Lior539 opened the pull request:
Context - https://github.com/PostHog/posthog.com/issues/6791

Please comment on the issue, so we have discussions in one place

benjackwhite reviewed (changes requested):
One orphaned link in the sidebar. Otherwise this is great!

<img width="269" alt="Screenshot 2023-09-25 at 08 33 36" src="https://github.com/PostHog/posthog/assets/2536520/528f16de-cbb2-4c4a-86a1-8cb5dd415836">

Lior539 commented:
Fix the orphaned side bar and made updates following @tiina303's [comment in the issue](https://github.com/PostHog/posthog.com/issues/6791#issuecomment-1731561492).

I have now also:
- Deleted the [monitoring with Grafana page](https://posthog.com/docs/self-host/configure/monitoring-with-grafana)
- Deleted the [Deploying Clickhouse using Atuinity page](https://posthog.com/docs/self-host/configure/using-altinity-cloud)
- Deleted more sections in the Troubleshooting page (are there more questions I need to delete?)

Is there anything else that needs to change? I think best to preview the vercel deployment to get a better picture of whats left

tiina303 commented on contents/docs/self-host/configure/securing-posthog.mdx:
```diff
@@ -69,14 +69,12 @@ make it is secure to threat actors. As every installation is unique, we provide
 it is then up to each instance admin to make sure all the best practices are followed.
 
-1. make sure your Kubernetes cluster is provisioned and configured using industries
+1. Make sure your Kubernetes cluster is provisioned and configured using industries
```
remove the `## Securing PostHog Deployments` that's tied to k8s. The rest of the envs are relevant for others too.

tiina303 commented on contents/docs/self-host/deploy/troubleshooting.md:
also remove 
1. `### How do I see logs for a pod?`
2. ### How do I connect to Postgres?

pod is a k8s concept (you can see `kubectl` there -> k8s not something one would use on docker compose

tiina303 reviewed (approved):
>  move some of this stuff to the internal repos as some of it may be useful for internal teams. I'll do that in a separate PR

thanks

Lior539 commented:
@joethreepwood this needs to be merged, and I also need to move the content to the internal repo as TIina suggested. I've deprioritized it for now in favor of focusing on blog content+tutorials. I plan to do it sometime this quarter, though.

If there's a particular need or urgency, I can prioritize it?

joethreepwood commented:
No urgency - just tidying up some old PRs and noticed this one hadn't moved in a while!
